pkg/gcp/machine_controller_util.go

```diff
@@ -81,6 +81,17 @@ func (ms *MachinePlugin) CreateMachineUtil(ctx context.Context, machineName stri
 		}
 	)
 
+	if providerSpec.GPUConfig != nil {
+		instance.GuestAccelerators = []*compute.AcceleratorConfig{
+			{
+				AcceleratorType:  fmt.Sprintf("projects/%s/zones/%s/acceleratorTypes/%s", project, zone, providerSpec.GPUConfig.AcceleratorType),
+				AcceleratorCount: providerSpec.GPUConfig.CountPerNode,
+			},
+		}
+		// need to make onHostMaintainence as `TERMINATE` as live-migration not supported for VMs with GPU attached
+		instance.Scheduling.OnHostMaintenance = "TERMINATE"
```

Comment:
You may want to store "TERMINATE" in a global constant.

Reply:
You are overwriting `OnHostMaintenance` to `TERMINATE` which is set as part of `MachineClass`. Would it make sense to do a validation and reject  the machine class creation request if `GPU` has been associated to the machine class and its `OnHostMaintenance` is set to `MIGRATE` so that there is no surprise that the `MachineClass` spec was accepted as is but then its `OnHostMaintenance` was silently overwritten?

Reply:
I see its assigned here in extension provider https://github.com/gardener/gardener-extension-provider-gcp/blob/98cd853353c2698f482ae7f75d34a23fbbfc9f35/pkg/controller/worker/machines.go#L176-L179

As of now I don't see any validation in extension-gcp related to `onHostMaintainence` which can block machineClass creation. 

I would do 3 things:
- will add it as part of PR which makes relevant GPU changes in extension-gcp.
- would add validation in machine-controller itself, so that machineClass is not processed even if its created with incorrect values
- will not set onHostMaintainence to `TERMINATE` explicitly.

Hope thats ok